pkg/insights/insightsclient/insightsclient.go

```diff
@@ -284,16 +284,67 @@ func (c *Client) Send(ctx context.Context, endpoint string, source Source) error
 
 // RecvReport perform a request to Insights Results Smart Proxy endpoint
 func (c Client) RecvReport(ctx context.Context, endpoint string) (*io.ReadCloser, error) {
+	resp, err := c.doGetRequest(ctx, endpoint, true)
+
+	if err != nil {
+		klog.Errorf("Unable to retrieve latest report for %s: %v", c.clusterVersion.Spec.ClusterID, err)
+		counterRequestRecvReport.WithLabelValues(c.metricsName, "0").Inc()
+		return nil, err
+	}
+
+	counterRequestRecvReport.WithLabelValues(c.metricsName, strconv.Itoa(resp.StatusCode)).Inc()
+	err = validateResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode == http.StatusOK {
+		return &resp.Body, nil
+	}
+
+	klog.Warningf("Report response status code: %d", resp.StatusCode)
+	return nil, fmt.Errorf("Report response status code: %d", resp.StatusCode)
+}
+
+// RecvOCMData sends a get request to the OCM API endpoint defined in the config
+func (c Client) RecvOCMData(ctx context.Context, endpoint string) ([]byte, error) {
+	resp, err := c.doGetRequest(ctx, endpoint, false)
+
+	if err != nil {
+		klog.Errorf("Unable to retrieve data from %s: %v", endpoint, err)
+		return nil, err
+	}
+
+	err = validateResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode == http.StatusOK {
+		data, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, err
+		}
+		return data, nil
+	}
+
+	klog.Warningf("OCM response status code: %d", resp.StatusCode)
+	return nil, fmt.Errorf("OCM response status code: %d", resp.StatusCode)
```

Comment:
Same nitpick as above. I just feel like this doesn't look very nice and it could result in some pretty confusing behavior if someone only changed one of the lines.